addons/spreadsheet_dashboard/models/spreadsheet_dashboard.py

```diff
@@ -15,8 +16,35 @@ class SpreadsheetDashboard(models.Model):
     is_published = fields.Boolean(default=True)
     company_id = fields.Many2one('res.company')
     group_ids = fields.Many2many('res.groups', default=lambda self: self.env.ref('base.group_user'))
+    favorite_user_ids = fields.Many2many(
+        'res.users', string='Favorite Users', help='Users who have favorited this dashboard'
+    )
+    is_favorite = fields.Boolean(
+        compute='_compute_is_favorite', string='Is Favorite',
+        help='Indicates whether the dashboard is favorited by the current user'
+    )
     main_data_model_ids = fields.Many2many('ir.model')
 
+    @api.depends('favorite_user_ids')
+    def _compute_is_favorite(self):
+        current_user_id = self.env.uid
+        for dashboard in self:
+            dashboard.is_favorite = current_user_id in dashboard.favorite_user_ids.ids
+
+    def action_toggle_favorite(self):
+        self.ensure_one()
+        try:
+            self.check_access_rights('read')
+            self.check_access_rule('read')
+        except AccessError:
+            raise AccessError(_("You cannot add or remove this dashboard from your favorites"))
+
+        current_user_id = self.env.uid
+        if current_user_id in self.sudo().favorite_user_ids.ids:
+            self.sudo().favorite_user_ids = [(3, current_user_id)]
```

Comment:
I think we have aliases for magic numbers now, to make them more readable. `from odoo import Command`